### PR TITLE
Winningest coach

### PR DESCRIPTION
**Pull request by sciencefixion**
If the stages are Red, Green, Refactor this is in the Green, but certainly could use a refactor coming up. I plan to break winningest_coach into some smaller helpers when I find out what I need in subsequent methods. I left comments just so it makes more sense because it's a bit of a mess now. I plan to change the setup for my tests to use the large game teams CSV I added, but the methods do pass all tests with the current setup. I welcome your feedback!

### CODE REVIEW

**Comment by sciencefixion** [comment]
@gabichuelas @evilaspaas1 @rrabinovitch :)

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -118,7 +118,49 @@ def lowest_scoring_home_team
...
+    game.result == "WIN"
+    end
+
+    # returns an array of coach name for each win
+    coach_wins = wins.map do |game|
+    game.head_coach
+    end
+
+    # creates a hash of number of season games won by coach
+    num_wins_by_coach = coach_wins.inject(Hash.new(0)) { |coach, count| coach[count] += 1; coach}
```
Comment (gabichuelas) [inline lib/stat_tracker.rb]: @sciencefixion you want to flip your block variables around. The first variable is the "accumulator", so it's whatever you want to name your hash that you are starting with, and the second variable is the item you are iterating over, which in this case would be a `head_coach`. 

So maybe your code should be:
```
num_wins_by_coach = coach_wins.inject(Hash.new(0)) { |num_wins_by_coach, coach| num_wins_by_coach[coach] += 1; num_wins_by_coach }
```

Obv that's a really long name for a block variable, but I wanted to illustrate with the terms you were already using what the block variables are representing.

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -118,7 +118,49 @@ def lowest_scoring_home_team
...
+
+    #return the winningest head_coach name as a string
+    coach_wins.max_by { |count| num_wins_by_coach[count] }
```
Comment (gabichuelas) [inline lib/stat_tracker.rb]: @sciencefixion taking into account my last comment, you'd change this to be `num_wins_by_coach[coach]` as the key in that hash should be a coach name.

**Review by gabichuelas — CHANGES_REQUESTED** [review]
@sciencefixion  Looks good, and clearly all your code works, but you'll want to change your variables a bit because they are not referencing what you think they're referencing!! (left comments per lines)

File: lib/stat_tracker.rb (hunk 2 above)
Comment (sciencefixion) [reply lib/stat_tracker.rb]: Made these changes! Thank you for your help!

File: lib/stat_tracker.rb (hunk 1 above)
Comment (sciencefixion) [reply lib/stat_tracker.rb]: Thanks for this. I was running on fumes last night when I was working on this. So glad to have your eyes on it!

**Review by araspaas — APPROVED** [review]
looks good to me.

**Review by gabichuelas — APPROVED** [review]
@sciencefixion Looks good and you say all tests pass, so I'll merge it!!

**Pull request merged**